session.go

```diff
@@ -1,97 +1,77 @@
 package bond
 
 import (
+	"database/sql"
+	"fmt"
 	"reflect"
 	"sync"
 
-	"upper.io/db"
+	"upper.io/db.v2"
+	"upper.io/db.v2/lib"
 )
 
 type Session interface {
-	db.Tx
+	lib.SQLDatabase
+
 	Store(interface{}) Store
 	Find(...interface{}) db.Result
 	Save(Model) error
 	Delete(Model) error
-	NewTransaction() (Session, error)
-	ContinueTransaction() (Session, error, bool)
+
+	BondTx(func(tx Session) error) error
 }
 
 type session struct {
-	db.Database
+	lib.SQLDatabase
 	stores     map[string]*store
 	storesLock sync.Mutex
 }
 
-func Open(adapter string, url db.ConnectionURL) (Session, error) {
-	conn, err := db.Open(adapter, url)
+// Open connects to a database.
+func Open(url db.ConnectionURL) (Session, error) {
```

Comment:
yea, exactly. the more minimal, the better. as well, I think its important bond is just a small extension, an interface-enhancer.. and doesnt hide all of upper, and having the connection be the first thing makes that also very clear